src/sublime_aio.py

```diff
@@ -1,20 +1,40 @@
 from __future__ import annotations
+
 import asyncio
 import atexit
+import functools
 import traceback
 from inspect import iscoroutinefunction
 from threading import Thread
-from typing import TYPE_CHECKING
-
-if TYPE_CHECKING:
-    from asyncio import Task
-    from concurrent.futures import Future
-    from typing import Any, Callable, Coroutine
+from typing import TYPE_CHECKING, overload
 
 import sublime
 import sublime_api
 import sublime_plugin
 
+if TYPE_CHECKING:
+    from collections.abc import Coroutine
+    from concurrent.futures import Future
+    from typing import Any, Callable, List, Tuple, TypeVar, Union
+
+    from typing_extensions import ParamSpec, TypeAlias
+
+    P = ParamSpec('P')
+    T = TypeVar('T')
+
+    BlankCoro: TypeAlias = Coroutine[object, None, None]
+    CompletionsList: TypeAlias = Union[List[str], List[Tuple[str, str]], List[sublime.CompletionItem]]
+    CompletionsReturnVal: TypeAlias = Union[
+        sublime.CompletionList,
+        Tuple[CompletionsList, sublime.AutoCompleteFlags],
+        CompletionsList,
+        None,
+    ]
```

Comment:
This one is causing type match errors as API expects `list[CompletionValue]`. It can be used here as `CompletionValue` is exported by `sublime` module in type checking mode.

Furthermore, 

1. as there's already a `sublime.CompletionList`, I'd avoid defining a type alias with nearly equal name.

2. a completion list can consist of items of mixed types (string, tuple and CompletionItem).

Hence the previous version was more correct.

Reply:
The previous version does not validate against LSP stubs, however. We'll need to decide which is more correct, ideally based on what ST actually supports, and adjust our own stubs (to eventually be pushed as a wheel to pypi or github) accordingly.

https://github.com/sublimelsp/LSP/blob/177ae9b25c8242e697fc96767cf32363329958f1/stubs/sublime.pyi#L3135

Reply:
I'd rather follow official API and update stubs. Not sure when they have been adjusted to upstream changes.

Reply:
LSPs stubs are probably what plugins would do, but those are incorrect with regards to what ST API supports/expects. Not sure how LSP leverages those stubs, but I see type errors when using this form.